components/content_settings/core/browser/brave_content_settings_pref_provider.cc

```diff
@@ -44,22 +44,22 @@ Rule CloneRule(const Rule& rule, bool reverse_patterns = false) {
 
 class BraveShieldsRuleIterator : public RuleIterator {
  public:
-  BraveShieldsRuleIterator(std::vector<Rule>::const_iterator iterator,
-                           std::vector<Rule>::const_iterator end)
-      : iterator_(iterator),
-        end_(end) {}
+  explicit BraveShieldsRuleIterator(std::vector<Rule> rules)
+      : rules_(std::move(rules)) {
+    iterator_ = rules_.begin();
```

Comment:
if we pass the vector now, why do we still need to save its iterator?

Reply:
because we still need to iterate over it

Reply:
I meant we can get the iterator from `rules_ ` when we need to iterate over it

Reply:
I'm not sure what you're proposing because it gets called repeatedly in `Next()`. How can we do that if we don't keep a reference to the iterator? Doing some kind of lazy initialization seems like it would just complicate the code for no benefit that I can see